src/openai/_utils/_logs.py

```diff
@@ -23,3 +25,15 @@ def setup_logging() -> None:
         _basic_config()
         logger.setLevel(logging.INFO)
         httpx_logger.setLevel(logging.INFO)
+
+
+class APIKeyFilter(logging.Filter):
+    @override
+    def filter(self, record: logging.LogRecord) -> bool:
+        if isinstance(record.args, dict) and "headers" in record.args:
+            if isinstance(record.args["headers"], dict):
+                logged_headers: Dict[str, Any] = record.args["headers"]
+                for header in logged_headers:
+                    if header.lower() in ["api-key", "authorization"]:
+                        logged_headers[header] = "<redacted>"
```

Comment:
nit: we have an `is_dict()` header that would make this slightly cleaner imo
```suggestion     
        if is_dict(record.args) and "headers" in record.args and is_dict(record.args["headers"]):
            headers = record.args["headers"]
            for header in headers:
                if str(header).lower() in ["api-key", "authorization"]:
                    headers[header] = "<redacted>"

```
and `from ._utils import is_dict`.

also a minor micro optimisation but I think we should extract the list of headers, e.g.
```py
SENSITIVE_HEADERS = {"api-key", "authorization"}
```

Reply:
also I haven't actually written a log filter before, is it bad that this mutates the original `headers` arg? i.e. should it be this?
```py
            headers = record.args["headers"] = {**record.args["headers"]}
```

Reply:
Updated with your revisions, thanks! The logging docs for `filter` permits in-place modification:

> If deemed appropriate, the record may be modified in-place.